packages/fxa-content-server/app/scripts/views/mixins/email-autocomplete-domains-mixin.js

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A mixin to autocomplete common domains from `datalist option` `value`s.
+ */
+import Template from 'templates/partial/email-autocomplete-domains.mustache';
+
+export const EMAIL_SELECTOR = 'input[type=email]';
+export const DATALIST_OPTIONS_SELECTOR = '#autocomplete-domain option';
+export const FOCUS_HACK_SELECTOR = '#focus-hack';
+
+export const DOMAINS = [
+  'gmail.com',
+  'outlook.com',
+  'hotmail.com',
+  'yahoo.com',
+  'qq.com',
+  'web.de',
+  'aol.com',
+  'mail.ru',
+  'icloud.com',
+  'gmx.de',
+  't-online.de',
+  'orange.fr',
+  'yandex.ru',
+  'yahoo.fr',
+  'live.com',
+  '163.com',
+  'msn.com',
+  'comcast.net',
+  'hotmail.co.uk',
+  'hotmail.fr',
+];
+
+export default {
+  events: {
+    [`keyup ${EMAIL_SELECTOR}`]: '_toggleDomainAutocomplete',
+  },
+
+  setInitialContext(context) {
+    const unsafeEmailAutocompleteDomainsHTML = this.renderTemplate(Template, {
+      options: DOMAINS,
+    });
+    context.set({
+      unsafeEmailAutocompleteDomainsHTML,
+    });
+  },
+
+  afterRender() {
+    this.input = this.$(EMAIL_SELECTOR);
+    this.datalistOptions = this.$(DATALIST_OPTIONS_SELECTOR);
+    this.focusHack = this.$(FOCUS_HACK_SELECTOR);
+    this.previousUsername = null;
+  },
+
+  /**
+   * Catch-all-browsers hack when datalist options are modified.
+   * Chrome otherwise shows all options when '@' is typed,
+   * Safari immediately goes to the dropdown preventing users
+   * from typing further, and FF will not update the dropdown on `keyup`
+   * and has a known bug with dynamic datalists
+   * https://bugzilla.mozilla.org/show_bug.cgi?id=1474137
+   */
+  _focusHack() {
+    this.focusHack.focus();
+    this.input.focus();
+  },
+
+  _toggleDomainAutocomplete() {
+    const inputValue = this.input.val();
+    const username = inputValue.substr(0, inputValue.indexOf('@'));
+
+    if (
+      inputValue.length >= 2 &&
+      inputValue.includes('@') &&
+      ((this.previousUsername !== null && username !== this.previousUsername) ||
+        !this.input.attr('list'))
+    ) {
+      // also trigger _before_ datalist option mods here because Chrome
+      // looks janky if text entered pulls up the autocomplete list
+      this._focusHack();
```

Comment:
I considered checking for user agent while working on this PR, but this line is the only place I'd need the check (for Chrome) since we actually need `_focusHack()` for other browsers in the other two spots that method is called. 🤷‍♀️